frontend/src/scenes/billing/BillingProduct.tsx

```diff
@@ -159,6 +162,19 @@ export const BillingProduct = ({ product }: { product: BillingProductV2Type }):
     } = useActions(billingProductLogic({ product }))
     const { reportBillingUpgradeClicked } = useActions(eventUsageLogic)
 
+    const [survey, setSurvey] = useState<Survey | null>()
+    const submitSurvey = useCallback(
+        (textAreaValue: string) => {
+            posthog.capture('survey sent', {
+                $survey_id: survey.id,
+                $survey_name: survey.name,
+                $survey_response: textAreaValue,
+            })
+            deactivateProduct(product.type)
+        },
+        [survey]
+    )
```

Comment:
We use [keajs](https://keajs.org/) (project by marius) for state management. It's a bit cumbersome but more reliable than using useState everywhere. You'll want to create actions in `billingProductLogic.ts` for `toggleSurveyModalOpen` and `submitSurvey`. Use a reducer to store the value of `isSurveyModalOpen` and react to the toggle action (similar to other modal examples in there). Use a listener for the submit action to do the event call. 

actually there is an `eventUsageLogic` where many events are captured from that would be good to use for consistency.